html/syntax/parsing/test.js

```diff
@@ -211,7 +211,12 @@ function test_fragment(iframe, t, test_id, uri_encoded_input, escaped_expected,
      container_elem = document.createElement(container);
   }
   container_elem.innerHTML = input_string;
-  var serialized_dom = test_serializer(container_elem);
+  var serialized_dom;
+  if (container_elem.namespaceURI === namespaces["html"] && container_elem.localName === "template") {
+    serialized_dom = test_serializer(container_elem.content);
+  } else {
+    serialized_dom = test_serializer(container_elem);
+  }
```

Comment:
Why not implement this in test_serializer?

Reply:
That feels like a bit of a layering violation; `test_serializer` literally just serializes the DOM node it's given. The special-case here is specific to how we run the fragment tests.